From: Russ Allbery <eagle@eyrie.org>
Date: Sun, 29 Dec 2013 12:32:00 -0800
Subject: Loosen user_mtime sanity check

One of the Debian buildds failed the previous sanity check of a
user mtime after 2000, possibly because the relevant file was
missing from the chroot.  Loosen the check to accept 0, but still
catch significantly invalid results.
---
 tests/server/basic-t.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/tests/server/basic-t.c b/tests/server/basic-t.c
index ff19f7d..a2131c1 100644
--- a/tests/server/basic-t.c
+++ b/tests/server/basic-t.c
@@ -55,14 +55,14 @@ is_sane_reply(struct lbcd_request *request, struct lbcd_reply *reply,
 
     /* Check the general system information. */
     now = time(NULL);
-    ok(ntohl(reply->boot_time) > 946713600,
+    ok((time_t) ntohl(reply->boot_time) > 946713600,
        "...boot time later than 2000-01-01");
     ok((time_t) ntohl(reply->boot_time) < now,
        "...boot time before the current time");
     ok(labs((time_t) ntohl(reply->current_time) - now) < 2,
        "...current time within two seconds");
-    ok(ntohl(reply->user_mtime) > 946713600,
-       "...user mtime later than 2000-01-01");
+    ok((time_t) ntohl(reply->user_mtime) >= 0,
+       "...user mtime later than 1970-01-01");
     ok((time_t) ntohl(reply->user_mtime) < now,
        "...user mtime before the current time");
     ok(ntohs(reply->l1) < 20000, "...1 minute load less than 200");
